File: src/main/java/by/architecture_map/belarus/service/ArchitecturalStyleService.java
package by.architecture_map.belarus.service;

import by.architecture_map.belarus.entity.ArchitecturalStyle;

import java.util.List;

public interface ArchitecturalStyleService {

    ArchitecturalStyle create(ArchitecturalStyle architecturalStyle);

    List<ArchitecturalStyle> findAll();

    ArchitecturalStyle find(int id);

    ArchitecturalStyle update(int id, ArchitecturalStyle updatedArchitecturalStyle);

    ArchitecturalStyle patchUpdate(int id, ArchitecturalStyle updatedArchitecturalStyle);

    void delete(int id);
}

File: src/main/kotlin/by/architecture_map/belarus/service/ArchitecturalStyleService.kt
package by.architecture_map.belarus.service

import by.architecture_map.belarus.entity.ArchitecturalStyle

interface ArchitecturalStyleService {

    fun create(architecturalStyle: ArchitecturalStyle): ArchitecturalStyle
    fun findAll(): List<ArchitecturalStyle>
    fun find(id: Int): ArchitecturalStyle
    fun update(id: Int, updatedArchitecturalStyle: ArchitecturalStyle): ArchitecturalStyle
    fun patchUpdate(id: Int, updatedArchitecturalStyle: ArchitecturalStyle): ArchitecturalStyle
    fun delete(id: Int)
}
